File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map <int,int> sum;
       for(int i = 0;i<nums.size();i++){
           int val=nums[i];
           int complement=target-val;
           auto it=sum.find(complement);
           if(it!=sum.end()){
            return {it->second, i};
               }
           sum[val]=i;
       }
        return {};
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapVal={}
        for i in range(len(nums)):
            val=target-nums[i]
            if val in mapVal:
                return [mapVal[val],i]
            mapVal[nums[i]]=i
        return